src/is/basicLegacyError.ts

```diff
@@ -0,0 +1,10 @@
+import { IBasicLegacyError } from 'src/types/error_types';
+
+/**
+ * Type guard to check if something is a subset of the interface BasicError.
+ *
+ * @param thing to check type of
+ */
+export function isBasicError(thing: unknown): thing is IBasicLegacyError {
```

Comment:
I would refactor this. I would put each error interface in its own file in `/src/types/errors` and then each one of these helper functions would be defined in the same file where the interface is defined. Is that possible?